File: problems_list/0023.合并K个升序链表/0023-合并K个升序链表.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* merge(ListNode* a, ListNode* b){
        if(!a || !b) return a?a:b;
        ListNode *dummy=new ListNode();
        ListNode *tmp=dummy, *ap=a, *bp=b;
        while(ap!=nullptr && bp!=nullptr){
            if(ap->val<bp->val){
                tmp->next=ap;
                ap=ap->next;
            }else{
                tmp->next=bp;
                bp=bp->next;
            }
            tmp=tmp->next;
        }
        tmp->next=(ap?ap:bp);
        return dummy->next;
    }

    ListNode* mergeKLists(vector<ListNode*>& lists) {
        int n=lists.size();
        ListNode* ans=nullptr;
        for(int i=0; i<n; i++){
            ans=merge(ans, lists[i]);
        }
        return ans;
    }
};

File: problems_list/0023.合并K个升序链表/0023-合并K个升序链表.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode[]} lists
 * @return {ListNode}
 */
var merge=function(la, lb){
    if(la===null || lb===null) return la?la:lb;
    let dummy=new ListNode(0);
    let cur=dummy, ap=la, bp=lb;
    while(ap && bp){
        if(ap.val<bp.val){
            cur.next=ap;
            ap=ap.next;
        }else{
            cur.next=bp;
            bp=bp.next;
        }
        cur=cur.next;
    }
    cur.next=(ap?ap:bp);
    return dummy.next;
}

var mergeKLists = function(lists) {
    let ans=null;
    for(let i=0; i<lists.length; i++){
        ans=merge(ans, lists[i]);
    }
    return ans;
};